### PR TITLE
US30 shelters statistics COMPLETED

### PR DESCRIPTION
**Pull request by robertorodriguez12**
Author: Roberto
Contributors: Roberto and Priya

US30: Shelter Statistics
As a visitor
When I visit a shelter's show page
I see statistics for that shelter, including:
- count of pets that are at that shelter
- average shelter review rating
- number of applications on file for that shelter

**Updates Needed:** Refactor of Shelter controller methods.

**View Updates:** We need a working welcome page (welcome's individual html); we need an established header and footer on all other webpages (THIS IS A PARTIAL!), we need a nav bar on ALL webpages (this belongs in application.html)

### CODE REVIEW

File: app/controllers/shelters_controller.rb [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
     @shelters = Shelter.all
   end
 
   def show
     @shelter = Shelter.find(params[:id])
     @reviews = @shelter.reviews
+    @pet_counter = @shelter.pets.count
+    @average_rating = @reviews.average(:rating)
+    @shelter_applications = Array.new
+    @shelter.pets.each do |pet|
```
Comment (priyapower) [inline app/controllers/shelters_controller.rb:14]: Awesome job! I think we can use a .reduce or a .map here to create our new array so we can call `.flatten!` on the end  of the do block like you were trying during pairing

File: app/controllers/shelters_controller.rb [hunk 2]
```diff
@@ -1,19 +1,36 @@
...
 
   def create
-    Shelter.create(shelter_params)
-    redirect_to '/shelters'
+    shelter = Shelter.new(shelter_params)
+    if shelter.save
+      redirect_to '/shelters'
+    else
+      quantity = empty_fields(params).count
+      flash[:shelter_fields_notice] = "Shelter Creation Warning: You are missing #{pluralize(quantity, "field")}: #{empty_fields_convert}"
```
Comment (priyapower) [inline app/controllers/shelters_controller.rb:31]: Here is an opportunity to use that empty_fields on ANY page that has warning about required fields - We should consider moving this to application controller and updating

File: app/controllers/shelters_controller.rb [hunk 3]
```diff
@@ -38,6 +60,31 @@ def destroy
   end
 
   private
+
+  def empty_fields(current_params)
```
Comment (priyapower) [inline app/controllers/shelters_controller.rb:64]: We can probably move these two methods to the application controller now

**Review by priyapower — COMMENTED** [review]
Left a few refactor notes. Looks awesome!

File: app/controllers/shelters_controller.rb (hunk 1 above)
Comment (robertorodriguez12) [reply app/controllers/shelters_controller.rb:14]: That is so true, something to explore during refactoring.

**Pull request merged**